### PR TITLE
Add checkpoint interface for custom checkpoints

### PR DESCRIPTION
**Pull request by harlow**
To allow other checkpoint backends we extract a checkpoint interface
which future checkpoints can implement.

* Create RedisCheckpoint to implement checkpoint interface
* Swap out hosie redis library for go-redis

Minor changes

* Allow configuration of Redis endpoint with env var `REDIS_URL`

Implements https://github.com/harlow/kinesis-connectors/issues/28

### CODE REVIEW

**Comment by harlow** [comment]
@jsam let me know if this looks OK to you. I've tested it locally and things are looking ok, but second set of eyes would be great.

**Comment by harlow** [comment]
If a bad REDIS_URL exists in the ENV the program will error out:

```
kinesis-connectors/examples/s3(hw-redis-config) $ REDIS_URL=bad-redis-url go run main.go -a test -s test
 ERROR[0000] Redis connection failed   app=test error=dial tcp: missing port in address bad-redis-url package=kinesis-connectors stream=test
exit status 1
```

File: redis_checkpoint_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		StreamName: "stream",
+		client:     client,
+	}
+
+	// set checkpoint
+	c.SetCheckpoint("shard_id", "testSeqNum")
+
+	// checkpoint exists
+	if val := c.CheckpointExists("shard_id"); val != true {
+		t.Fatalf("checkpoint exists expected true, got %s", val)
```
Comment (jsam) [inline redis_checkpoint_test.go]: CheckpointExists returns bool, so `val` will be bool. Can we use %t verb to format print on bool correctly? =))

Comment (harlow) [inline redis_checkpoint_test.go]: good catch. updated in 34f0b1e

**Comment by jsam** [comment]
🚀 🚀 

look's great! thank you so much!

let's 🚢

**Pull request merged**